python/ray/autoscaler/_private/kubernetes/config.py

```diff
@@ -71,8 +71,14 @@ def fillout_resources_kubernetes(config):
         return config
     node_types = copy.deepcopy(config["available_node_types"])
     for node_type in node_types:
-        container_data = node_types[node_type]["node_config"]["spec"][
-            "containers"][0]
+
+        node_config = node_types[node_type]["node_config"]
+        # The next line is for compatibility with configs like
+        # kubernetes/example-ingress.yaml,
+        # cf. KubernetesNodeProvider.create_node().
+        pod = node_config.get("pod", node_config)
+        container_data = pod["spec"]["containers"][0]
+
```

Comment:
Is this relevant to this PR? I don't understand why the field change from:
`node_types[node_type]["node_config"]["spec"]["containers"][0]`
to
`node_types[node_type]["node_config"]["pod"]["spec"]["containers"][0]`

Reply:
Reverting this change causes `test_autoscaler_yaml.py` to fail for me locally (so I assume it would fail in the CI as well).
Specifically, `kubernetes/example-ingress.yaml` fails validation with a key error because it specifies its node configs differently --
it does
```
head_node:
    pod: <k8s pod config>
    service: <k8s service config>
    ...
```
while the rest of the examples do 
```
head_node:
    <k8s pod config>
```

The other option is to delete `kubernetes/example-ingress.yaml` which would be fine too.

This same logic appears in KubernetesNodeProvider:
https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/kubernetes/node_provider.py#L102

Reply:
Is this backward compatible?

Reply:
Yeah, this change is basically an awkward fix to preserve backwards-compatibility.
I'll run the operator unit test with the changes to make sure I didn't break anything.